modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -442,7 +442,10 @@ public void visit(BLangBlockStmt blockNode) {
         SymbolEnv blockEnv = SymbolEnv.createBlockEnv(blockNode, this.env);
 
         for (BLangStatement stmt : blockNode.stmts) {
-            addLineNumberInfo(stmt);
+            if (stmt.getKind() != NodeKind.TRY && stmt.getKind() != NodeKind.CATCH
```

Comment:
what if there transaction ?

Reply:
Transaction scenarios are not tested with debugger yet, will test them and improve this transaction scenarios later

Reply:
created a issue to track this https://github.com/ballerinalang/ballerina/issues/3591